packages/account/src/BiconomySmartAccount.ts

```diff
@@ -256,25 +256,57 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
     return "0x";
   }
 
-  async buildUserOp(transactions: Transaction[], overrides?: Overrides, skipBundlerGasEstimation?: boolean): Promise<Partial<UserOperation>> {
+  async buildUserOp(
+    transactions: Transaction[],
+    overrides?: Overrides,
+    skipBundlerGasEstimation?: boolean,
+    paymasterServiceData?: SponsorUserOperationDto,
+  ): Promise<Partial<UserOperation>> {
     this.isInitialized();
     const to = transactions.map((element: Transaction) => element.to);
     const data = transactions.map((element: Transaction) => element.data ?? "0x");
     const value = transactions.map((element: Transaction) => element.value ?? BigNumber.from("0"));
     this.isProxyDefined();
 
+    // Queue promises to fetch independent data.
+    const nonceFetchPromise = (async () => {
+      let nonce = BigNumber.from(0);
+      try {
+        nonce = await this.nonce();
+      } catch (error) {
+        // Not throwing this error as nonce would be 0 if this.nonce() throw exception, which is expected flow for undeployed account
+      }
+      return nonce;
+    })();
+
+    const gasFeeValues = {
+      maxFeePerGas: overrides?.maxFeePerGas,
+      maxPriorityFeePerGas: overrides?.maxPriorityFeePerGas,
+    };
+    // Estimate max fee per gas and max priority fee per gas
+    const getGasFeeValues = (async () => {
```

Comment:
make these functions (nonceFetchPromise and gasFeeValues) outside of buildUserOp as inline functions can cause performance issues as every time the outerfunction is called, the new instance of inner function is created in memory.

Rename functions to getNonce and getGasFeeValues